velox/exec/Unnest.cpp

```diff
@@ -220,13 +220,52 @@ RowVectorPtr Unnest::generateOutput(
       // Construct unnest column using Array elements wrapped using above
       // created dictionary.
       auto unnestBaseArray = currentDecoded.base()->as<ArrayVector>();
-      outputs[outputsIndex++] = identityMapping
-          ? unnestBaseArray->elements()
-          : wrapChild(
-                numElements,
-                elementIndices,
-                unnestBaseArray->elements(),
-                nulls);
+      auto elements = unnestBaseArray->elements();
+
+      if (elements->typeKind() == TypeKind::ROW && !legacyUnnest_) {
+        // In this case the array is un-nested into multiple columns, one for
+        // each child type of the row.
+        // The array row elements could be encoded however. In that case, each
+        // child array accessed needs to be encoded with the array row elements
+        // encoding and then wrapped with the un-nest encoding for final output.
+        DecodedVector rowDecoded(*elements);
+        auto rowVector = rowDecoded.base()->as<RowVector>();
+
+        auto rowIsIdentity = rowDecoded.isIdentityMapping();
+        auto numRowElements = rowDecoded.size();
+
+        // These need to be constructed because the DecodedVector indices/nulls
```

Comment:
Would you check if DecodedVector::wrap API can be used here?

Reply:
Thanks @mbasmanova. DecodedVector.wrap works as well.